AK/Optional.h

```diff
@@ -186,7 +186,7 @@ requires(!IsLvalueReference<T>) class [[nodiscard]] Optional<T> : public Optiona
     }
 
     AK_MAKE_CONDITIONALLY_COPYABLE(Optional, <T>);
```

Comment:
Commit message says it "should be considered undefined behavior to access a moved-from Optional". But that's not how moved-from objects work. It must be a 'valid but unspecified state'. It's a bug, to be sure, but never UB.